include/universal/utility/generateClosurePlots.hpp

```diff
@@ -0,0 +1,450 @@
+ #include <universal/utility/generateClosurePlots.hpp> 
+ 
+ namespace sw { namespace universal {
+ 
+ 
+ template<typename NumberType, char Op>
+ struct OperationStruc {
+     static constexpr char getOperationChar() { return Op; }
+ 
+     static constexpr std::string_view getOperationString() {
+         if (Op == '+') return "addition";
+         if (Op == '-') return "subtraction";
+         if (Op == '*') return "multiplication";
+         if (Op == '/') return "division";
+         return "unknown";
+     }
+ 
+     static NumberType primary(NumberType a, NumberType b) {
+         if (Op == '+') return a + b;
+         if (Op == '-') return a - b;
+         if (Op == '*') return a * b;
+         if (Op == '/') return a / b;
+         return NumberType(0);
+     }
+ 
+     static NumberType inverse(NumberType a, NumberType b) {
+         if (Op == '+') return a - b;
+         if (Op == '-') return a + b;
+         if (Op == '*') return a / b;
+         if (Op == '/') return a * b;
+         return NumberType(0);
+     }
+ 
+     NumberType executeOperation(NumberType a, NumberType b) const {
+         return primary(a, b);
+     }
+ 
+     NumberType executeInverseOperation(NumberType a, NumberType b) const {
+         return inverse(a, b);
+     }
+ };
+ 
+ struct NumberSystemStats {
+     NumberSystemStats() : total{ 0 }, nars_and_nans { 0 }, exact{ 0 }, approximate{ 0 }, overflow{ 0 }, underflow{ 0 }, saturate{ 0 } {}
+     unsigned long total;
+     unsigned long nars_and_nans;
+     unsigned long exact;
+     unsigned long approximate;
+     unsigned long overflow;
+     unsigned long underflow;
+     unsigned long saturate;
+     unsigned long absoluteError;
+     unsigned long relativeError;
+     unsigned long relativeLogError;
+ };
+ 
+ 
+ /**
+  * helper function for systemEvaluators: 
+  * 
+  *  3 error metrics are calculated:
+  *      i. absolute error (absErr)
+  *      ii. relative error (relativeErr)
+  *      iii. relative log error (relativeLogErr)
+  *              - this value is normalized with the logarithmic min max normalization function (MinMaxLogNormalization)
+  * 
+  *      All of these functions are found in error.hpp
+  */
+ template<typename NumberType>
+ void calculateError(const std::string& result, 
+                     double vcDouble, double targetVal, 
+                     double& absErr, double& relativeErr, double& relativeLogErr,
+                     NumberType maxpos, NumberType minpos){
+ 
+         if (result == "NAR"){
+             absErr = std::numeric_limits<double>::infinity();
+             relativeErr = std::numeric_limits<double>::infinity();
+             relativeLogErr = std::numeric_limits<double>::infinity();
+         }
+         
+         else if (result == "Exact"){
+             absErr = 0;
+             relativeErr = 0;
+             relativeLogErr = 0;
+         }
+ 
+         else{
+             absErr = std::abs(sw::universal::AbsoluteError(vcDouble, targetVal));
```

Comment:
since you define this function in the namespace sw::universal, you do not need to explicitely reference them through a fully articulated name. Simply calling AbsoluteError() will do.